docs/specification.md

```diff
@@ -1,153 +1,146 @@
+## Methods
 
-### builder_getPayloadHeaderV1
+### `builder_setFeeRecipientV1`
 
 #### Request
 
-- method: `builder_getPayloadHeaderV1`
+- method: `builder_setFeeRecipientV1`
 - params:
-  1. `payloadId`: `DATA`, 8 Bytes - Identifier of the payload build process
+  1. `feeRecipient`: `DATA`, 20 Bytes - Address of account which should receive fees.
+  2. `timestamp`: `QUANTITY`, uint64 - Unix timestamp of announcement.
+  2. `publicKey`: `DATA`, 48 Bytes - Public key of validator.
+  3. `signature`: `DATA`, 96 Bytes - Signature over `feeRecipient`.
 
 #### Response
 
-- result: [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/bellatrix/beacon-chain.md#executionpayloadheader)
+- result: `null`
 - error: code and message set in case an exception happens while getting the payload.
 
-### relay_getPayloadHeaderV1
+#### Specification
+1. Builder software **MUST** verify `signature` is valid under `publicKey`.
+2. Builder software **MUST** respond to requests where `timestamp` is before the latest announcement from the validator or more than +/- 10 seconds from the current time with `-32006: Invalid timestamp`.
+3. Builder software **MUST** store `feeRecipient` in a map keyed by `publicKey`.
+
+### `builder_getBlindExecutionPayloadV1`
 
 #### Request
 
-- method: `relay_getPayloadHeaderV1`
+- method: `builder_getBlindExecutionPayloadV1`
 - params:
-  1. `payloadId`: `DATA`, 8 Bytes - Identifier of the payload build process
+  1. `hash`: `DATA`, 32 Bytes - Hash of block which the validator intends to use as the parent for its proposal.
 
 #### Response
 
-- result: [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
+- result: `object`
+    - `payload`: [`BlindExecutionPayloadV1`](#blindexecutionpayloadv1).
+    - `value`: `DATA`, 32 Bytes - the change in wei balance of the `feeRecipient` account.
+    - `publicKey`: `DATA`, 48 Bytes - the public key associated with the builder.
+    - `signature`: `DATA`, 96 Bytes - BLS signature of the builder over `payload` and `value`.
 - error: code and message set in case an exception happens while getting the payload.
 
-### relay_proposeBlindedBlockV1
+#### Specification
+1. Builder software **SHOULD** respond immediately with the `payload` that increases the `feeRecipient`'s balance by the most.
+2. Builder software **MUST** return `-32001: Unknown hash` if the block identified by `hash` does not exist.
+3. Builder software **MUST** return `-32002: Unknown validator` if the recovered validator public key has not been mapped to a `feeRecipient`.
+
+### `builder_getExecutionPayloadV1`
 
 #### Request
 
-- method: `relay_proposeBlindedBlockV1`
+- method: `builder_getExecutionPayloadV1`
 - params:
-  1. [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock)
-  2. [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
+  1. `block`: `DATA`, arbitray length - SSZ encoded [`BeaconBlock`][beacon-block].
```

Comment:
This has to be a `BlindedBeaconBlock` (an otherwise full beacon block with a blinded execution payload) because the CL has no knowledge of transactions at this point, just the transactions root

Reply:
Absolutely, my mistake - was just fixing that.